From 88d72e00a71dc9c48f34dd77f05b624cde97b2f7 Mon Sep 17 00:00:00 2001
From: Wang, Yi A <yi.a.wang@intel.com>
Date: Tue, 30 Apr 2013 13:52:46 +0800
Subject: Awesomeplayer: when the clip is not seekable, send out the notification

BZ: 48604

original patch:
R3: 37219

local TS clip is not seekable, when we seek, even it can not be seeked,
we should notifythe app seek complete,so app will update the progressbar.

Category: aosp improvement
Domain: Video.Media-local playback
Origin: internal
Upstream-Candidate: yes

Orig-Change-Id: I9241d6ce9408609109bbb77d50b463f9a2cd8250
Change-Id: I41a54cf38e8e14e215701276e5d2a922ca954c82
Signed-off-by: Weian Chen <weian.chen@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
Signed-off-by: Yuanjun Huang <yuanjun.huang@intel.com>
---
 media/libstagefright/AwesomePlayer.cpp |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/AwesomePlayer.cpp b/media/libstagefright/AwesomePlayer.cpp
index 130207d..f18dd14 100644
--- a/media/libstagefright/AwesomePlayer.cpp
+++ b/media/libstagefright/AwesomePlayer.cpp
@@ -1405,6 +1405,8 @@ status_t AwesomePlayer::seekTo(int64_t timeUs) {
     if (mExtractorFlags & MediaExtractor::CAN_SEEK) {
         Mutex::Autolock autoLock(mLock);
         return seekTo_l(timeUs);
+    } else {
+        notifyListener_l(MEDIA_SEEK_COMPLETE);
     }
 
     return OK;
-- 
1.7.4.1

